File: php/spec/float.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('float', function() {
  describe('with \'42\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::float('42'))->toBe(42.0);
    });
  });
  
  describe('with \'-42\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::float('-42'))->toBe(-42.0);
    });
  });
  
  describe('with \'42.0\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::float('42.0'))->toBe(42.0);
    });
  });
  
  describe('with \'42,0\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::float('42,0'); })->toThrow();
    });
  });
  
  describe('with \'4 2.0\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::float('4 2.0'); })->toThrow();
    });
  });
  
  describe('with \'fortytwo\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::float('fortytwo'); })->toThrow();
    });
  });
});

File: ruby/spec/float.spec.rb
require 'enotype'

describe 'float' do
  context 'with \'42\'' do
    it 'produces the expected result' do
      expect(Enotype::float('42')).to be(42.0)
    end
  end
  
  context 'with \'-42\'' do
    it 'produces the expected result' do
      expect(Enotype::float('-42')).to be(-42.0)
    end
  end
  
  context 'with \'42.0\'' do
    it 'produces the expected result' do
      expect(Enotype::float('42.0')).to be(42.0)
    end
  end
  
  context 'with \'42,0\'' do
    it 'raises an error' do
      expect { Enotype::float('42,0') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'4 2.0\'' do
    it 'raises an error' do
      expect { Enotype::float('4 2.0') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'fortytwo\'' do
    it 'raises an error' do
      expect { Enotype::float('fortytwo') }.to raise_error(RuntimeError)
    end
  end
end